### ISSUE TITLE
Lakka 3.5 Bluetooth Controllers Refuse to Auto Connect

### ISSUE THREAD

**Issue by PhasecoreX** [libretro/Lakka-LibreELEC#1528]
- Which version of Lakka are you using?
  - RPi3.aarch64-3.5
- What system hardware are you using?
  - Raspberry Pi 3B+
- What did you do?
  - Updated from 3.4 to 3.5, pressed the A button on my WiiU pro controller to turn it on.
- What did you expect to happen?
  - Expected the controller to connect automatically, as it has always done in versions prior.
- What happened instead?
  - The controller player LEDs blinks once and doesn't connect.

When running `bluetoothctl`, every time I turn the controller on (by tapping power/start/A) I see this message pop up:
```
[CHG] Device 40:F5:07:D8:E3:2B Connected: no
```
So it seems like the controller is connecting to the Pi, then whatever is running on the Pi that would accept the connection just doesn't work, and then the controller turns off. If I have the Pi turned off, turning the controller causes the player LEDs to blink for 30 seconds, as it is searching for the Pi to connect to. That leads me to believe something on the Pi is actively refusing the connection for some reason (and causing that log message above).

From both `bluetoothctl` as well as Lakkas bluetooth menu, I can manually connect to the controller. With `bluetoothctl`, simply using the `connect <mac_address>`, and then turning the controller on while it's attempting to connect allows the controller to connect successfully. With the Lakka bluetooth menu, I need to select the controller, and then press the pairing button on the controller. So I am still able to use them, just that it's pretty inconvenient (since I need a keyboard to actually get to that menu).

Looking in `/storage/.cache/bluetooth/B8:57:AB:61:C3:5C`, I see my four controller MAC addresses as 4 folders. The info file in each shows them as Trusted and not Blocked. They worked for 3.4, haven't touched them since (or ever).

Any help with this would be greatly appreciated! And if you need any other info let me know.

**Comment by ToKe79** [comment]
This might be due to the added driver for Xbox Wireless gamepads (`xpadneo`) - you can disable this driver by:
```
echo "blacklist xpadneo" > /storage/.config/modprobe.d/blacklist-xpadneo.conf
```
Please, let us know if this solves your issue.

**Comment by PhasecoreX** [comment]
I have not paired these controllers with any other device, they have exclusively been used with this Raspberry Pi.

However I have removed them all from Lakka with the space bar as you mentioned, restarted Lakka, and then paired one of the controllers. Upon rebooting, it again would not automatically connect. The same observations in my original post are still present (one LED blink, bluetoothctl log message, etc).

I even went and removed the entire `/storage/.cache/bluetooth` folder, rebooted, and paired a controller. Upon rebooting again, still doesn't auto connect.

**Comment by PhasecoreX** [comment]
Bluetooth service is enabled. For pairing the controller, I used the Lakka Bluetooth menu. Using the menu, I can pair the controller and it immediately connects after the pairing process. However once the controller is turned off and then back on, the controller will not automatically connect.

I did just now try the `bluetoothctl` method of pairing the controller and then trusting it. But when I turn the controller on, it again will not automatically connect. Same message as before in the log:

```
[CHG] Device 40:F5:07:D8:E3:2B Connected: no
```

**Comment by ToKe79** [comment]
To rule out / isolate the issue, can you, please, roll back to 3.4 and confirm, this issue is not present? Thank you!

**Comment by PhasecoreX** [comment]
I have downgraded to `RPi3.aarch64-3.4`, paired a controller with the Lakka bluetooth menu, rebooted the Pi, and upon turning on the controller it connects successfully! I paired my other 3 controllers and they all connect automatically as well.

**Comment by ToKe79** [comment]
thank you, I will try to find a fix, maybe it is really related to the xpadneo driver, maybe not...

**Comment by jdalmanza** [comment]
Same issue on pi4 upgraded from 3.4 to 3.5.1
Also using the WiiU Pro Controller -- configured via GUI and not paired to any other devices other than this pi4 since lakka v3.0

please let me know if you need any logs.

**Comment by ToKe79** [comment]
On a side note - I had similar issue today on RPi400. I looked at the folders and noticed that the `/storage/.cache` folder and subfolders had weird owner - uid 1023 (no such uid in Lakka). Then I recalled that I had the drive connected to my Android TV box (I was copying over files) which might have messed up the ownership. After fixing the ownership (`chown root:root .cache -R`) pairing started working.
